### DISCUSSION TITLE
A question about internal motor control settings (EV3)

### DISCUSSION THREAD

**Discussion by Tacvud** [pybricks/support#211] [General]
First of all, I really appreciate the Pybricks platform and I am grateful to its authors that made it to its current level. I am still a Pybrick beginner but with every project learn more and more.

In my project focused on designing a robotic arm, I started to experiment with motor settings, I mean those available via the motor.control.limits(). I am interested to learn how the internal speed limit is set, if it depends on the type of the attached (medium, large) motor, on the gear ratios, or whatever. My motivation is to find out, if I can move the speed limits higher, of course, in a safe way.

Thanks.

**Comment by laurensvalk** [comment]
> My motivation is to find out, if I can move the speed limits higher, of course, in a safe way.

Increasing the speed limit is fine; your motor will never receive a higher voltage than the battery voltage.

**Reply by Tacvud** [reply]
Sorry, I should have explained my question in more detail. 
My motor (EV3 medium) is declared as follows:
motorGrip = Motor(Port.B, positive_direction=Direction.COUNTERCLOCKWISE, gears=[1,24]) 

When I check its limits by: motor['Grip'].control.li﻿mits())
I receive: (42, 67, 100).     Funny, that a couple of days before I received different numbers: (33, 67, 100)

When I compare the first number to the medium motor specs (260 rpm at https://www.philohome.com/motors/motorcomp.htm), take into account the gear ratio, a higher speed limit in deg/s should be possible.

That is why I wanted to learn how the internal speed limit is set.

**Comment by laurensvalk** [comment]
> a couple of days before I received different numbers:

["When you specify a gear train, all motor commands and settings are automatically adjusted to account for the resulting gear ratio."](https://docs.pybricks.com/en/latest/ev3devices.html#pybricks.ev3devices.Motor)

This ensures that all speeds work the same way, which can be quite convenient.

> a higher speed limit in deg/s should be possible.

The default speed limits are a bit lower than what they theoretically could be. This is intentional, so that performance is roughly the same whether your batteries are fully charged or not.

You can set them higher if you like; only at some point it won't make your robot go any faster.

**Reply by Tacvud** [reply]
Thank you, laurensvalk, for fast responses.

What is really strange is that I received the different speed limit values (once 33 and later 42) for the exactly same robot assembly and the same code. That is why I was confused.

Moreover, my motivation to increase the speed of the gripper is that the gripper is significantly slower (in terms of deg/s) than the other moving parts, which complicates those robot actions where all the motors run in parallel, and any further increase of the speed parameter in motor[m].run_target(speed, ...]) does not help. Therefore, I checked the Pybricks documentation, found that motor.control.li﻿mits() statements and tried to explore how it works.

**Reply by laurensvalk** [reply]
Between motors of the same type, the defaults should be the same. EV3 Medium motors can go a bit faster, so their defaults are a bit higher. Could that be it?

Occasionally, a large motor gets (incorrectly) detected as a medium motor. To see if this is the case, you can check the device manager on the EV3 screen menu.

**Reply by Tacvud** [reply]
I checked how the motors are identified. Now, all are correct. But maybe, it the past one or more were identified wrongly and that was the reason for strangely set limits. Thanks for explanation.
